zebra-consensus/src/transaction/tests.rs

```diff
@@ -693,13 +696,179 @@ async fn mempool_request_with_unmined_output_spends_is_accepted() {
     );
 
     tokio::time::sleep(POLL_MEMPOOL_DELAY * 2).await;
+    // polled before AwaitOutput request and after a mempool transaction with transparent outputs
+    // is successfully verified
     assert_eq!(
         mempool.poll_count(),
         2,
-        "the mempool service should have been polled twice, \
-     first before being called with an AwaitOutput request, \
-     then again shortly after a mempool transaction with transparent outputs \
-     is successfully verified"
+        "the mempool service should have been polled twice"
+    );
+}
+
+#[tokio::test]
+async fn skips_verification_of_block_transactions_in_mempool() {
+    let mut state: MockService<_, _, _, _> = MockService::build().for_prop_tests();
+    let mempool: MockService<_, _, _, _> = MockService::build().for_prop_tests();
+    let (mempool_setup_tx, mempool_setup_rx) = tokio::sync::oneshot::channel();
+    let verifier = Verifier::new(&Network::Mainnet, state.clone(), mempool_setup_rx);
+    let verifier = Buffer::new(verifier, 1);
+
+    mempool_setup_tx
+        .send(mempool.clone())
+        .ok()
+        .expect("send should succeed");
+
+    let height = NetworkUpgrade::Canopy
+        .activation_height(&Network::Mainnet)
+        .expect("Canopy activation height is specified");
+    let fund_height = (height - 1).expect("fake source fund block height is too small");
+    let (input, output, known_utxos) = mock_transparent_transfer(
+        fund_height,
+        true,
+        0,
+        Amount::try_from(10001).expect("invalid value"),
+    );
+
+    // Create a non-coinbase V4 tx with the last valid expiry height.
```

Comment:
Optional: I think it will be good to use v5 transactions in all new tests as eventually v4 will be probably be deprecated.

Reply:
Good idea, practically free extra test coverage, applied in https://github.com/ZcashFoundation/zebra/pull/8951/commits/af9001dd91fdec765ec4137882bd8831bf0402db.